File: server/src/main/java/ro/jtonic/handson/hackathon/server/ApplicationProps.java
package ro.jtonic.handson.hackathon.server;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.stereotype.Component;

/**
 * Created by Antonel Ernest Pazargic on 26/03/2017.
 *
 * @author Antonel Ernest Pazargic
 */
@EnableConfigurationProperties
@ConfigurationProperties(prefix = "app")
@Component
public class ApplicationProps {
    private String firstName;
    private String lastName;

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

}

File: server/src/main/kotlin/ro/jtonic/handson/hackathon/server/ApplicationProps.kt
package ro.jtonic.handson.hackathon.server

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

/**
 * Created by Antonel Ernest Pazargic on 23/02/2017.
 * @author Antonel Ernest Pazargic
 */
@Component
data class ApplicationProps(
        @Value("\${app.first-name}") val firstName: String,
        @Value("\${app.last-name}") val lastName: String
)